src/OrchardCore/OrchardCore/Shell/ShellDescriptorFeaturesManager.cs

```diff
@@ -36,8 +36,11 @@ public class ShellDescriptorFeaturesManager : IShellDescriptorFeaturesManager
         {
             var featureEventHandlers = ShellScope.Services.GetServices<IFeatureEventHandler>();
 
-            var enabledFeatureIds = _extensionManager.GetFeatures()
+            var enabledFeatures = _extensionManager.GetFeatures()
                 .Where(f => shellDescriptor.Features.Any(sf => sf.Id == f.Id))
+                .ToHashSet();
```

Comment:
@jtkech I made the change. Any reason why `ToArray()` instead of `ToHashSet()`?

Reply:
Okay thanks

Both prevent re-evaluation, ToHashSet is useful when we then use Add/Remove and it will compare to existing items in an optimized way, works with strings and classes that are IEquatable otherwise the comparizon is by ref (object pointers are compared) which works too if a given featureInfo is always related to the same object instance, which I think it is the case here as they are always returned from the extensionManager collection but ...